input/system/selfhosted/otel_handler.go

```diff
@@ -0,0 +1,210 @@
+				return true
+			}
+		} else {
+			prefixedLogger.PrintWarning("Pod specification for OTel server not valid (need zero or one / separator): \"%s\", skipping log record\n", server.Config.LogOtelK8SPod)
+			return true
+		}
+	}
+
+	if server.Config.LogOtelK8SLabels != "" {
+		selectors := strings.Split(server.Config.LogOtelK8SLabels, ",")
+		for _, selector := range selectors {
+			parts := k8sSelectorRegexp.FindStringSubmatch(selector)
+			if parts != nil {
+				selKey := parts[1]
+				selEq := parts[2] == "=" || parts[2] == "=="
+				selNotEq := parts[2] == "!="
+				selValue := parts[3]
+				v, ok := k8sLabels[selKey]
+				if ok {
+					if (selEq && v != selValue) || (selNotEq && v == selValue) {
+						return true
+					}
+				}
+			} else {
+				prefixedLogger.PrintWarning("Label selector for OTel server not valid: \"%s\", skipping log record\n", server.Config.LogOtelK8SLabels)
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func setupOtelHandler(ctx context.Context, server *state.Server, rawLogStream chan<- SelfHostedLogStreamItem, parsedLogStream chan state.ParsedLogStreamItem, prefixedLogger *util.Logger) error {
+	otelLogServer := server.Config.LogOtelServer
+	tz := server.GetLogTimezone()
+	go func() {
+		http.HandleFunc("/v1/logs", func(w http.ResponseWriter, r *http.Request) {
+			b, err := io.ReadAll(r.Body)
+			if err != nil {
+				prefixedLogger.PrintError("Could not read otel body")
+			}
+			logsData := &otlpLogs.LogsData{}
+			if err := proto.Unmarshal(b, logsData); err != nil {
+				prefixedLogger.PrintError("Could not unmarshal otel body")
+			}
+			for _, r := range logsData.ResourceLogs {
+				for _, s := range r.ScopeLogs {
+					for _, l := range s.LogRecords {
+						var logger string
+						var record *common.KeyValueList
+						var kubernetes *common.KeyValueList
+						hasErrorSeverity := false
+						if l.Body.GetKvlistValue() != nil {
+							for _, v := range l.Body.GetKvlistValue().Values {
+								if v.Key == "logger" {
+									logger = v.Value.GetStringValue()
+								}
+								if v.Key == "record" {
+									record = v.Value.GetKvlistValue()
+								}
+								if v.Key == "kubernetes" {
+									kubernetes = v.Value.GetKvlistValue()
+								}
+								if v.Key == "error_severity" {
+									hasErrorSeverity = true
+								}
+							}
+							// TODO: Support other logger names (this is only tested with CNPG)
+							if logger == "postgres" {
+								// jsonlog wrapped in K8s context (via fluentbit)
+								logLine, detailLine := logLineFromJsonlog(record, tz)
+								if skipDueToK8sFilter(kubernetes, server, prefixedLogger) {
+									continue
+								}
+
+								parsedLogStream <- state.ParsedLogStreamItem{Identifier: server.Config.Identifier, LogLine: logLine}
+								if detailLine != nil {
+									parsedLogStream <- state.ParsedLogStreamItem{Identifier: server.Config.Identifier, LogLine: *detailLine}
+								}
+							} else if logger == "" && hasErrorSeverity {
```

Comment:
Why are we requiring `error_severity`? It might be good to add a comment here or above.

Reply:
Good question. This part is written by Lukas and actually don't have good context. This is about the following part:

> Note this requires the log output to be jsonlog (Postgres 15+), with optional K8s context added, or plain logs without any additional context.


It is "plan logs without any additional context". I actually haven't tested this case. I think `error_severity` is somewhat "must be there" JSON key as a JSON format logs, therefore if that key exists, we can assume that it's a Postgres log? @lfittl does it sound right? 
https://www.postgresql.org/docs/current/runtime-config-logging.html#RUNTIME-CONFIG-LOGGING-JSONLOG

Reply:
(I updated the comment assuming that it's correct, but happy to update)